Doc/library/typing.rst

```diff
@@ -820,6 +821,8 @@ The module defines the following classes, functions and decorators:
       assert employee.id == 3
 
    Fields with a default value must come after any fields without a default.
+   Note that default values are incompatible with ``__slots__`` which are
+   needed to save the space take by instance dictionaries.
```

Comment:
See my comment above. It is already said that subclassing ``NamedTuple`` is equivalent to calling ``collections.namedtuple``. Maybe we need to document that it is not just equivalent, but it is waht actually happens at runtime.